xapian-applications/omega/docs/newformat.rst

```diff
@@ -0,0 +1,174 @@
+   ::
+
+     convert_to_utf8(text, "ISO-8859-1");
+
+2. Once you have the library and the mimetype, it is time to modify the code. We have to create a new handler, which is the process used by omindex to access to the library. In order to do it, we have to create a file ‘handler_yourlibrary.cc’ that includes ‘handler.h’ and gives a definition to function ``extract`` (there are some examples at xapian-applications/omega such as 'handler_poppler.cc'). For this, inside the function body you will use the library to get the necessary information and store it in the corresponding arguments
+   ::
+
+     bool
+     extract(const std::string& filename,
+             std::string& dump,
+             std::string& title,
+             std::string& keywords,
+             std::string& author,
+             std::string& pages,
+             std::string& error)
+
+  Error messages should be written in ``error`` (you wont be able to write in the standard input/output) and you should fill the rest of the arguments with the information extracted from the document (It is not necessary to fill in all the arguments).
+
+  You can get more information about this function at 'xapian-applications/omega/handler.h'.
+
+3. After the handler is implemented, the build system must be updated. In particular, it is necessary to modify 'configure.ac' and 'makefile.am'.
+
+   In 'configure.ac', we need to check if the library is available using ``PKG_CHECK_MODULES`` and define ``HAVE_LIBRARY`` if we can use it. It is highly recommended to follow the example of the other handlers.
+   Some macros that you might find useful are ``AC_CHECK_HEADERS``, ``AC_DEFINE``, ``AC_COMPILE``, ``AC_LINK_IFELSE``
+   ::
+
+     dnl check if libraryname is available.
+     PKG_CHECK_MODULES([LIBRARY], [library_pkg-config], [
+         AC_DEFINE(HAVE_LIBRARY, 1, [Define HAVE_LIBRARY if the library is available])
+         OMINDEX_MODULES="$OMINDEX_MODULES omindex_yourlibrary"],[ ])
+
+   In makefile.am,  we should add 'omindex_yourlibrary' to ``ExtraProg`` and define the variables ``omindex_yourlibrary_SOURCES``, and  ``omindex_yourlibrary_LDADD`` and ``omindex_yourlibrary_CPPFLAGS`` if they are necessary. ``library_pkg-config`` must be replace by the name of the pkg-config file of the library (you can find it by looking for a .pc file inside the directory of your library).
+
+   If we want to add 'omindex_poppler', we will get
+   ::
+
+     dnl check if poppler is available.
+     PKG_CHECK_MODULES([POPPLER], [poppler-cpp], [
+         AC_DEFINE(HAVE_POPPLER, 1, [Define HAVE_POPPLER if the library is available])
+         OMINDEX_MODULES="$OMINDEX_MODULES omindex_poppler"],[ ])
```

Comment:
Except that's not how we probe for poppler, so this isn't really a good example.